clicommand/agent_start.go

```diff
@@ -900,8 +900,14 @@ var AgentStartCommand = cli.Command{
 			registerReq.Name = strings.ReplaceAll(cfg.Name, "%spawn", strconv.Itoa(i))
 
 			if cfg.SpawnWithPriority {
-				l.Info("Assigning priority %s for agent %d", strconv.Itoa(i), i)
-				registerReq.Priority = strconv.Itoa(i)
+				p := i
+				if experiments.IsEnabled("descending-spawn-priority") {
+					// This experiment helps jobs be assigned across all hosts
+					// in cases where the value of --spawn varies between hosts.
+					p = -i
```

Comment:
Why just use negative values starting with `-1` instead of subtracting from what `priority` is set in the configuration? With something in the lines of:
```
p = cfg.Priority - i + 1
```

I would expect it to behave like mentioned in the PR https://github.com/buildkite/agent/pull/1967:
```
# New behaviour: Start 5 agents with priorities [42, 41, 40, 39, 38].
buildkite-agent start --build-path=~/tmp --spawn=5 --spawn-with-priority=descending --priority=42
```